src/py/py_spline.cpp

```diff
@@ -304,42 +304,27 @@ py::array_t<double> PySpline::ParametricBounds() const {
   return pbounds;
 }
 
-py::array_t<double> PySpline::GrevilleAbscissae() const {
+py::list PySpline::GrevilleAbscissae(const bool allow_duplicates) const {
   // prepare output
   std::vector<int> cmr(para_dim_);
   Core()->SplinepyControlMeshResolutions(cmr.data());
-  const auto n_positions = [&]() {
-    int prod{cmr[0]};
-    for (std::size_t i{1}; i < cmr.size(); i++) {
-      prod *= cmr[i];
-    }
-    return prod;
-  }();
 
-  // Calculate Abscissae
-  py::array_t<double> greville_abscissae(n_positions * para_dim_);
-  double* greville_abscissae_ptr =
-      static_cast<double*>(greville_abscissae.request().ptr);
+  // Initialize return values
+  py::list list_of_greville_knots(para_dim_);
 
-  int offset{1};
-  int repetitions{n_positions};
   for (int i{}; i < para_dim_; i++) {
-    repetitions /= cmr[i];
-    std::vector<double> abscissae(cmr[i]);
-    Core()->SplinepyGrevilleAbscissae(abscissae.data(), i);
-    int ii{i};
-    for (int j{}; j < repetitions; j++) {
-      for (int l{}; l < cmr[i]; l++) {
-        for (int k{}; k < offset; k++) {
-          greville_abscissae_ptr[ii] = abscissae[l];
-          ii += para_dim_;
-        }
-      }
-    }
-    offset *= cmr[i];
+    // Calculate Abscissae
+    py::array_t<double> greville_abscissae(cmr[i]);
+    double* greville_abscissae_ptr =
+        static_cast<double*>(greville_abscissae.request().ptr);
+    Core()->SplinepyGrevilleAbscissae(greville_abscissae_ptr,
+                                      i,
+                                      allow_duplicates);
+    greville_abscissae.resize({cmr[i]});
```

Comment:
Same size resize?

Reply:
I thought this was required to fix shape in buffer. I'll erase that